src/apps/lwaftr/ipv6_apps.lua

```diff
@@ -147,3 +150,65 @@ function Fragmenter:push ()
       end
    end
 end
+
+-- TODO: handle any NS retry policy code here
+function NDP:new(conf)
+   local o = setmetatable({}, {__index=NDP})
+   o.conf = conf
+   -- TODO: verify that the src and dst ipv6 addresses and src mac address
+   -- have been provided, in pton format.
+   if not conf.dst_eth then
+      o.ns_pkt = ndp.form_ns(conf.src_eth, conf.src_ipv6, conf.dst_ipv6)
+      o.do_ns_request = true
+   else
+       o.do_ns_request = false
+   end
+   o.dst_eth = conf.dst_eth -- Intentionally nil if to request by NS
+   o.all_local_ipv6_ips = conf.all_ipv6_addrs
+   return o
+end
+
+function NDP:push()
+   local isouth, osouth = self.input.south, self.output.south
+   local inorth, onorth = self.input.north, self.output.north
+   if self.do_ns_request then
+      self.do_ns_request = false -- TODO: have retries, etc
+      transmit(osouth, packet.clone(self.ns_pkt))
+      -- TODO: do unsolicited neighbor advertisement on start and on
+      -- configuration reloads?
+      -- This would be an optimization, not a correctness issue
+   end
+   for _=1,link.nreadable(isouth) do
+      local p = receive(isouth)
+      if ndp.is_ndp(p) then
+         if not self.dst_eth and ndp.is_solicited_neighbor_advertisement(p) then
+            local dst_ethernet = ndp.get_dst_ethernet(p, {self.conf.dst_ipv6})
+            if dst_ethernet then
+               self.conf.dst_eth = dst_ethernet
```

Comment:
Can we avoid updating the config, and instead update self.dst_eth or so?

Other than this comment, this module looks really nice :-))